src/main/java/frc/robot/RobotContainer.java

```diff
@@ -50,41 +49,119 @@ public RobotContainer() {
         configureBindings();
     }
 
-    private void configureBindings() {
-        drive.registerTelemetry(logger::telemeterize);
-        drive.setDefaultCommand(new DriveWithJoysticks(drive, leftJoystick, rightJoystick));
+    private void configureSubsystems() {
+        // TODO: Potentially migrate to Constants.mode
+        if (Robot.isReal()) {
+            if (FeatureFlags.runDrive) {
+                drive =
+                        new PhoenixDrive(
+                                PhoenixDriveConstants.DrivetrainConstants,
+                                PhoenixDriveConstants.FrontLeft,
+                                PhoenixDriveConstants.FrontRight,
+                                PhoenixDriveConstants.BackLeft,
+                                PhoenixDriveConstants.BackRight);
+                logger = new Telemetry(6);
+            }
+            if (FeatureFlags.runVision) {
+                tagVision = new VisionLocalizer(new CameraContainerReal(VisionConstants.cameras));
+            }
+            if (FeatureFlags.runIntake) {
+                intakeSubsystem = new IntakeSubsystem(new IntakeNEOVortex());
+            }
+            if (FeatureFlags.runScoring) {
+                scoringSubsystem =
+                        new ScoringSubsystem(new ShooterIOTalonFX(), new AimerIORoboRio());
+            }
+        } else {
+            if (FeatureFlags.simulateDrive) {
+                drive =
+                        new PhoenixDrive(
+                                PhoenixDriveConstants.DrivetrainConstants,
+                                PhoenixDriveConstants.FrontLeft,
+                                PhoenixDriveConstants.FrontRight,
+                                PhoenixDriveConstants.BackLeft,
+                                PhoenixDriveConstants.BackRight);
+
+                logger = new Telemetry(6);
+            }
+            if (FeatureFlags.simulateVision) {
+                if (logger != null) {
+                    tagVision =
+                            new VisionLocalizer(
+                                    new CameraContainerSim(
+                                            VisionConstants.cameras, logger::getModuleStates));
+                } else {
+                    // TODO: Maybe try to spoof vision sim without drive telemetry by giving it all
+                    // zeros
+                    throw new NullPointerException("Vision simulation depends on drive!");
+                }
+            }
+            if (FeatureFlags.simulateIntake) {
```

Comment:
See minh's PR for why I think we shouldn't have separate feature flags for sim and robot. 

Long story short: this will cause config management problems and we should just get better at turning things on and off with JSON in a future PR